Formula/pyside.rb

```diff
@@ -63,17 +63,17 @@ def install
     # Install python scripts into pkgshare rather than bin
     inreplace "sources/pyside-tools/CMakeLists.txt", "DESTINATION bin", "DESTINATION #{pkgshare}"
 
-    args = std_cmake_args + [
-      "-DCMAKE_CXX_COMPILER=#{ENV.cxx}",
-      "-DCMAKE_PREFIX_PATH=#{Formula["qt"].opt_lib}",
-      "-DPYTHON_EXECUTABLE=#{which(python3)}",
-      "-DBUILD_TESTS=OFF",
-      "-DNO_QT_TOOLS=yes",
-      "-DCMAKE_INSTALL_RPATH=#{lib}",
-      "-DFORCE_LIMITED_API=yes",
-    ]
-
-    system "cmake", "-S", ".", "-B", "build", *args
+    # Avoid shim reference
+    inreplace "sources/shiboken6/ApiExtractor/CMakeLists.txt", "${CMAKE_CXX_COMPILER}", ENV.cxx
```

Comment:
Main diff is removing `-DCMAKE_CXX_COMPILER=#{ENV.cxx}` from CMake args and using above `inreplace` instead as `CMAKE_CXX_COMPILER` was bypassing shim (https://github.com/Homebrew/homebrew-core/pull/114598#discussion_r1015376123).

`pyside@2` inreplaces to `/usr/bin/c++` instead of `ENV.cxx` so not sure if that is better choice.

Corresponding code only runs on Linux. If `CMAKE_CXX_COMPILER` wasn't defined, it would use `"clang++"`. With change, it will use `"g++-11"`:
- https://code.qt.io/cgit/pyside/pyside-setup.git/tree/sources/shiboken6/ApiExtractor/CMakeLists.txt#n97
- https://code.qt.io/cgit/pyside/pyside-setup.git/tree/sources/shiboken6/ApiExtractor/clangparser/compilersupport.cpp#n302